### DISCUSSION TITLE
Strange Issue with views

### DISCUSSION THREAD

**Discussion by FraserElectronics** [AvaloniaUI/Avalonia#14283] [Q&A]
I created a simple Avalonia application for Raspberry Pi using the instructions provided here: [https://docs.avaloniaui.net/docs/guides/platforms/rpi/running-on-raspbian-lite-via-drm](url)

That worked great so, according to the documentation, I changed the XAML in MainView.axaml and rebuilt the solution. When I ran it on the desktop, the changes I have made to MainView were shown but on the Raspberry Pi, it was still showing the old MainView layout.

When I open MainSingleView.axaml in Visual Studio 2022 Preview I get:

![image](https://github.com/AvaloniaUI/Avalonia/assets/19693425/0071f503-adce-4fea-ab38-fd89283d1aa6)

When I open MainWindow.axaml in Visual Studio I get exactly the same BUT, this is what MainView.axaml actually looks like:

![image](https://github.com/AvaloniaUI/Avalonia/assets/19693425/6dc18fa0-e072-4cbf-a8d0-d4cb6fc41a43)

The axaml code for MainSingleView is:

```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             xmlns:views="clr-namespace:AvaOven.Views"
	     xmlns:vm="clr-namespace:AvaOven.ViewModels"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="480"
             x:Class="AvaOven.Views.MainSingleView">
	<views:MainView/>
</UserControl>
```

and the code in MainWindow.axaml is much the same (i.e. Calls <views:MainView>).

I am going mad ?!

Andy

**Comment by thevortexcloud** [comment]
Have you checked the files were actually updated on the Pi? Did you check if you copied the files from the correct place? I have run into problems like this before caused by me accidentally using an old build directory (EG using the .NET 7.0 folder instead of .NET 8.0)

**Reply by FraserElectronics** [reply]
Yes, I delete the files on the Pi before I upload them - but those screenshots are from Visual Studio. It is like Avalonia has cached the original axaml code somewhere. I can't figure it out 😒

Andy

**Reply by thevortexcloud** [reply]
Have you tried building it on the Pi? It will at least tell you if it's a deployment/Visual Studio problem.

**Reply by FraserElectronics** [reply]
Well, I am not sure what is going on but I changed the axaml code in MainView.axaml and rebuilt the solution and both MainSingleView and MainWindow updated to reflect the changes. I downloaded to the Pi and that worked. I then changed the axaml again and rebuilt and again the changes were correctly reflected in the other views and running on the Pi also showed these changes !
So, I am not at all sure what happened but it 'seems' to be working now !

Andy

**Reply by FraserElectronics** [reply]
@thevortexcloud Thanks for your help - hopefully it will continue work from here onwards 🤞

Andy

**Reply by thevortexcloud** [reply]
I am glad you solved it!
